packages/components/virtual-list/src/builders/buildGrid.ts

```diff
@@ -503,6 +575,30 @@ const createGrid = ({
         }
       }
 
+      // horizontal
+      const hScrollbar = h(Scrollbar, {
+        ref: 'hScrollbar',
+        clientSize: width,
+        layout: 'horizontal',
+        onScroll: onHorizontalScroll,
+        ratio: (width * 100) / estimatedTotalWidth,
+        scrollFrom: states.scrollLeft / (estimatedTotalWidth - width),
+        total: totalRow,
+        visible: true,
+      })
+
+      // vertical
+      const vScrollbar = h(Scrollbar, {
+        ref: 'vScrollbar',
+        clientSize: height,
+        layout: 'vertical',
+        onScroll: onVerticalScroll,
+        ratio: (height * 100) / estimatedTotalHeight,
+        scrollFrom: states.scrollTop / (estimatedTotalHeight - height),
+        total: totalColumn,
+        visible: true,
+      })
+
       const InnerNode = [
         h(
           Inner as VNode,
```

Comment:
Why use `as VNode`?

Reply:
```ts
    Argument of type 'VNodeTypes' is not assignable to parameter of type 'DefineComponent<{ style: any; ref: string; }, {}, {}, ComputedOptions, MethodOptions, ComponentOptionsMixin, ComponentOptionsMixin, ... 4 more ..., { ...; }>'.
      Type 'string' is not assignable to type 'DefineComponent<{ style: any; ref: string; }, {}, {}, ComputedOptions, MethodOptions, ComponentOptionsMixin, ComponentOptionsMixin, ... 4 more ..., { ...; }>'.
        Type 'string' is not assignable to type 'ComponentPublicInstanceConstructor<{ $: ComponentInternalInstance; $data: {}; $props: Partial<{ style: any; }> & Omit<Readonly<{ style?: unknown; ref?: unknown; } & { style: any; } & { ref?: string | undefined; }> & { ...; } & VNodeProps & AllowedComponentProps & ComponentCustomProps, "style">; ... 10 more ...; $wat...'.
```

Reply:
Because `Inner` and `Window` is resolved by `resolveDynamicComponent` method, which returns a type as `VNodeType` this type is synthetic by `string, VNode, DefineComponent<T>`, but `h` method provides various overloads which did not meet the type constrain.

Reply:
Maybe we can move type assert to `const Inner = resolveDynamicComponent(innerElement)`